scripts/setup/get-icons.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+const { exec, execSync } = require('child_process');
+const fs = require('fs');
+const chalk = require('chalk');
+const path = require('path');
+
+const fileExists = fs.existsSync('./.resources.json');
+if (!fileExists) {
+  console.log('File ./.resources.json not exists');
+  return;
+}
+
+const resourcesData = JSON.parse(fs.readFileSync('./.resources.json'));
+const iconFolder = 'public/assets/icons/';
+
+const callback = (error, stdout, stderr) => {
+  if (error) {
+    console.error(chalk`{red exec error: ${error}}`);
+    return;
+  }
+
+  if (stderr) {
+    console.error(`stderr: ${stderr}`);
+    return;
+  }
+
+  console.log(`stdout: ${stdout}`);
+  console.log(chalk`{green The icons have been successfully obtained. Check public/assets/icons/}`);
+};
+
+if (resourcesData.hasOwnProperty('repoUrl') || resourcesData.hasOwnProperty('path')) {
+  exec(`[ -d "${iconFolder}" ] && rm -rf ${iconFolder}`);
+  exec(`mkdir -p ${iconFolder}`);
+}
+
+if (resourcesData.hasOwnProperty('repoUrl')) {
+  if (!resourcesData.folder) {
+    execSync(
+      `git clone "${resourcesData.repoUrl}" --depth=1 --single-branch --branch ${resourcesData.branch} ${iconFolder} -o repo`,
+      {
+        stdio: [0, 1, 2], // we need this so node will print the command output
+        cwd: path.resolve(__dirname, '../../'), // path to where you want to save the file
+      },
+    );
+    return;
+  }
+  const tempFolder = 'tmp';
+
+  exec(`[ -d "${tempFolder}" ] && rm -rf ${tempFolder}`);
+  exec(`mkdir -p ${tempFolder}`);
+
+  execSync(
+    `git clone "${resourcesData.repoUrl}" --no-checkout  --depth=1 --single-branch --branch ${resourcesData.branch} ${tempFolder} -o repo`,
+    {
+      stdio: [0, 1, 2],
+      cwd: path.resolve(__dirname, '../../'),
+    },
+  );
+
+  execSync(
+    `git sparse-checkout init --cone &&
```

Comment:
`sparse-checkout` is great but its only available in 2.25. ubuntu 20.04 does have it but 18.04 is on 2.17. It is still too new so we should have a fallback for older git versions.

Reply:
Thanks for the comments! Oh ok! Well, I refactored a bit the `get-icons` file, so we can add multiple implementations for cloning a specific folder. 
Now we're asking for the git version and for git =>25  we are going to use the new version of `sparse-checkout`. And for  git < 25 we are going to use the old version of `sparse-checkout `(available from git `version 1.17`).